src/support_sphere/lib/logic/cubit/login_state.dart

```diff
@@ -33,8 +36,14 @@ final class LoginState extends Equatable {
       password: password ?? this.password,
       status: status ?? this.status,
       isValid: isValid ?? this.isValid,
+      isAllFieldsFilled: isAllFieldsFilled ?? this.isAllFieldsFilled,
       errorMessage: errorMessage ?? this.errorMessage,
       showPassword: showPassword ?? this.showPassword,
     );
   }
+  
+  // Check if all fields are filled
+  bool checkAllFieldsFilled() {
```

Comment:
This check is more of a business logic and I think it is best practice to place this within the cubit instead of the state. The logic would be that this check should run against the current state. You'd need some sort of event that emits a change in this `isAllFieldsFilled`.

1. You can use a bloclistener (https://bloclibrary.dev/flutter-bloc-concepts/) on the login form / signup form to check what the current state is
2. From there you can either modify the state by emitting an event or simply keeping track of just the button state within the widget